src/shared_modules/indexer_connector/src/indexerConnector.cpp

```diff
@@ -172,6 +172,34 @@ static void builderBulkDelete(std::string& bulkData, std::string_view id, std::s
     bulkData.append("\n");
 }
 
+static void builderDeleteByQuery(std::string& bulkData, const std::vector<std::string>& agentIds)
+{
+    bulkData.append(R"({
+  "query": {
+    "bool": {
+      "filter": {
+        "terms": {
+          "agent.id": [)");
+
+    // Append each agent ID to the JSON array
+    for (const auto& agentId : agentIds)
+    {
+        bulkData.append(R"(")");
+        bulkData.append(agentId);
+        bulkData.append(R"(")");
+        bulkData.append(",");
+    }
+    bulkData.pop_back(); // Remove the last comma
+
+    bulkData.append(R"(]
+        }
+      }
+    }
+  }
+})");
+    bulkData.append("\n");
+}
```

Comment:
We can simplify this:
```suggestion
static void builderDeleteByQuery(nlohmann::json& bulkData, const std::string& agentId)
{
    bulkData["query"]["bool"]["filter"]["terms"]["agent.id"].push_back(agentId);
}
```

Reply:
Tested [here](https://godbolt.org/z/WfqMjfMzY)

Reply:
I like the approach, and I am going to commit to it. Thanks for the idea!